src/tinyformat.h

```diff
@@ -124,6 +124,17 @@
 #ifndef TINYFORMAT_H_INCLUDED
 #define TINYFORMAT_H_INCLUDED
 
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#if defined(HAVE_W_ZERO_AS_NULL_POINTER_CONSTANT)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wzero-as-null-pointer-constant"
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wzero-as-null-pointer-constant"
```

Comment:
With gcc 9.3.0:
```
./tinyformat.h:134: warning: ignoring #pragma clang diagnostic [-Wunknown-pragmas]
  134 | #pragma clang diagnostic push
      | 
./tinyformat.h:135: warning: ignoring #pragma clang diagnostic [-Wunknown-pragmas]
  135 | #pragma clang diagnostic ignored "-Wzero-as-null-pointer-constant"
      | 
```

Reply:
Might be better to do with clang-tidy or so externally instead of maiming the source code?

Reply:
> Might be better to do with clang-tidy

Have done this in #24971.